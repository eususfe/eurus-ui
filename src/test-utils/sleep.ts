/*
 * @Author: jiangxd
 * @Date: 2022-09-23 14:06:55
 * @LastEditTime: 2022-09-23 14:07:04
 * @LastEditors: jiangxd
 * @Description:
 * @FilePath: /eurus-ui/src/test-utils/sleep.ts
 */
const sleep = (time = 0) => new Promise(resolve => setTimeout(resolve, time));

export default sleep;
