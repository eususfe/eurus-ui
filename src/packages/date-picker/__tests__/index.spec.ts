import { mount } from '@vue/test-utils';
import { nextTick } from 'vue';
import EDatePicker from '..';
import EuSelectDown from '../../select-down';

describe('EDatePicker ', () => {

  it('should work with `modelValue` props', async () => {

    const wrapper = mount(EDatePicker, {
      props: {
        modelValue: undefined
      },
    });
    wrapper.setProps({ modelValue: '2023-02-26' });
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();

    expect(wrapper.findComponent(EuSelectDown).find('.eu-select-down-single').text()).toBe('2023-02-26');

    wrapper.setProps({ modelValue: '2023-02-27' });
    await nextTick();
    expect(wrapper.findComponent(EuSelectDown).find('.eu-select-down-single').text()).toBe('2023-02-27');
  });

});
