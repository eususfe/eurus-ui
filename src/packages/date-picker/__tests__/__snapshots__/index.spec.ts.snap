// Vitest Snapshot v1

exports[`EDatePicker  > should work with \`modelValue\` props 1`] = `
"<div class=\\"eu-select-down eu-date-picker\\" type=\\"date\\">
  <div class=\\"eu-select-down-control eu-select-down-control-md\\" role=\\"listbox\\" tabindex=\\"0\\">
    <div class=\\"eu-select-down-single\\">2023-02-26</div><input class=\\"eu-select-down-control-input\\">
    <!----><span class=\\"eu-icon down-icon translate-icon\\" role=\\"link\\" tabindex=\\"0\\" style=\\"font-size: 20px;\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 32 32\\" class=\\"eu-icon eu-icon-chevron-down\\"><path fill=\\"currentColor\\" d=\\"M16 22 6 12l1.4-1.4 8.6 8.6 8.6-8.6L26 12z\\"></path></svg></span>
  </div>
  <transition-stub name=\\"slide-toggle\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
    <div class=\\"eu-select-down-panel\\" style=\\"display: inline-block;\\" role=\\"listbox\\" tabindex=\\"0\\">
      <div class=\\"eu-select-down-panel-wrapper\\">
        <ul class=\\"panel\\">
          <div class=\\"eu-date-picker-header\\" role=\\"link\\" tabindex=\\"0\\">
            <div class=\\"arrow-left\\">
              <!----><span class=\\"eu-icon arrow\\" role=\\"link\\" tabindex=\\"0\\" style=\\"font-size: 18px; color: #848b99;\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 32 32\\" class=\\"eu-icon eu-icon-chevron-left\\"><path fill=\\"currentColor\\" d=\\"M10 16 20 6l1.4 1.4-8.6 8.6 8.6 8.6L20 26z\\"></path></svg></span>
            </div>
            <div class=\\"date-content\\">2023-2</div>
            <div class=\\"arrow-right\\"><span class=\\"eu-icon arrow\\" role=\\"link\\" tabindex=\\"0\\" style=\\"font-size: 18px; color: #848b99;\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 32 32\\" class=\\"eu-icon eu-icon-chevron-right\\"><path fill=\\"currentColor\\" d=\\"M22 16 12 26l-1.4-1.4 8.6-8.6-8.6-8.6L12 6z\\"></path></svg></span>
              <!---->
            </div>
          </div>
          <div class=\\"eu-date-picker-body\\">
            <table>
              <thead class=\\"eu-date-picker-week-bar\\">
                <tr>
                  <th>一</th>
                  <th>二</th>
                  <th>三</th>
                  <th>四</th>
                  <th>五</th>
                  <th>六</th>
                  <th>日</th>
                </tr>
              </thead>
              <tbody class=\\"eu-date-picker-day-list\\">
                <tr>
                  <td class=\\"disable-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">30</div>
                  </td>
                  <td class=\\"disable-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">31</div>
                  </td>
                  <td class=\\"day-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">1</div>
                  </td>
                  <td class=\\"day-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">2</div>
                  </td>
                  <td class=\\"day-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">3</div>
                  </td>
                  <td class=\\"day-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">4</div>
                  </td>
                  <td class=\\"day-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">5</div>
                  </td>
                </tr>
                <tr>
                  <td class=\\"day-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">6</div>
                  </td>
                  <td class=\\"day-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">7</div>
                  </td>
                  <td class=\\"day-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">8</div>
                  </td>
                  <td class=\\"day-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">9</div>
                  </td>
                  <td class=\\"day-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">10</div>
                  </td>
                  <td class=\\"day-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">11</div>
                  </td>
                  <td class=\\"day-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">12</div>
                  </td>
                </tr>
                <tr>
                  <td class=\\"day-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">13</div>
                  </td>
                  <td class=\\"day-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">14</div>
                  </td>
                  <td class=\\"day-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">15</div>
                  </td>
                  <td class=\\"day-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">16</div>
                  </td>
                  <td class=\\"day-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">17</div>
                  </td>
                  <td class=\\"day-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">18</div>
                  </td>
                  <td class=\\"day-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">19</div>
                  </td>
                </tr>
                <tr>
                  <td class=\\"day-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">20</div>
                  </td>
                  <td class=\\"day-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">21</div>
                  </td>
                  <td class=\\"day-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">22</div>
                  </td>
                  <td class=\\"day-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">23</div>
                  </td>
                  <td class=\\"day-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">24</div>
                  </td>
                  <td class=\\"day-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">25</div>
                  </td>
                  <td class=\\"day-item dot\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">26</div>
                  </td>
                </tr>
                <tr>
                  <td class=\\"day-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">27</div>
                  </td>
                  <td class=\\"day-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">28</div>
                  </td>
                  <td class=\\"disable-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">1</div>
                  </td>
                  <td class=\\"disable-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">2</div>
                  </td>
                  <td class=\\"disable-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">3</div>
                  </td>
                  <td class=\\"disable-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">4</div>
                  </td>
                  <td class=\\"disable-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">5</div>
                  </td>
                </tr>
                <tr>
                  <td class=\\"disable-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">6</div>
                  </td>
                  <td class=\\"disable-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">7</div>
                  </td>
                  <td class=\\"disable-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">8</div>
                  </td>
                  <td class=\\"disable-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">9</div>
                  </td>
                  <td class=\\"disable-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">10</div>
                  </td>
                  <td class=\\"disable-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">11</div>
                  </td>
                  <td class=\\"disable-item\\" role=\\"gridcell\\">
                    <div class=\\"item-value\\">12</div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <div class=\\"eu-date-picker-footer\\" role=\\"button\\" tabindex=\\"0\\">今天</div>
        </ul><span class=\\"down-arrow\\"></span>
      </div>
    </div>
  </transition-stub>
</div>"
`;
